src/www/scripts/app.js

```diff
@@ -563,29 +563,29 @@ module.exports = (function() {
             try {
                 let localResult = await getLocalConfig();
 
-                getRemoteConfig().then((remoteResult) => {
-                    let updateConfig = async () => {
-                        await synchronizePackage(sourceUri, destinationUri);
-                        window.location.reload();
-                    }
+                let remoteResult = await getRemoteConfig();
 
-                    if (remoteResult.cachebust !== localResult.cachebust) {
-                        if (onAppUpdateAvailableFn) {
-                            onAppUpdateAvailableFn(updateConfig);
-                        } else {
-                            navigator.notification.confirm(__("An update is ready. Do you want to download it? (this may take a few moments)"),
-                                (buttonIndex) => buttonIndex === 1 && updateConfig(),
-                                __("Update ready"),
-                                [__("Yes"), __("No, update later")]
-                            );
-                        }
+                let updateConfig = async () => {
+                    await synchronizePackage(sourceUri, destinationUri);
+                    window.location.reload();
+                };
+
+                if (remoteResult.cachebust !== localResult.cachebust) {
+                    if (onAppUpdateAvailableFn) {
+                        onAppUpdateAvailableFn(updateConfig);
+                    } else {
+                        navigator.notification.confirm(__("An update is ready. Do you want to download it? (this may take a few moments)"),
+                            (buttonIndex) => buttonIndex === 1 && updateConfig(),
+                            __("Update ready"),
+                            [__("Yes"), __("No, update later")]
+                        );
                     }
-                }).catch((e) => {
-                    console.log('Unable to retrieve components.json');
-                });
+                }
 
                 return [ localResult, resourcesDirectory];
             } catch (e) {
+                console.log('Unable to retrieve components.json');
```

Comment:
getRemoteConfig

Reply:
Then `getRemoteConfig` at line `:589` should also be in a `try-catch`, correct?

Reply:
Don't think so, because if that fails, the whole synchronization should fail (so we can let the exception bubble up). However, the logic in the enableOffline try/catch block is incorrect at the moment. We have to take a look at it.